course02/homework01/danielellis/main.cc

```diff
@@ -0,0 +1,29 @@
+#include "linkedlist.h"
+
+#include <cassert>
+
+int main()
+{
+#ifdef NDEBUG
+#error Compile the code in debug mode!
+#endif
+
+	LinkedList l;
+	assert(l.Size() == 0);
+
+	l.Append(1);
+	assert(l.At(0) == 1);
+	assert(l.Size() == 1);
+
+	for (int i = 0; i < 100; ++i)
+	{
+		l.Append(i);
+		assert(l.At(i+1) == i);
+	}
+
+	assert(l.Size() == 101);
+	assert(l.At(1) == 0);
+	assert(l.At(50) == 49);
+
```

Comment:
You are not testing the copy of the list; and that's probably why you didnt catch the bug related to the shared_ptr passed by value instead of reference.